src/components/screens/CollectionView.vue

```diff
@@ -0,0 +1,411 @@
+                {{ editor.firstName }} {{ editor.lastName }}
+              </a>
+            </div>
+          </ul>
+          <p v-else>No editors</p>
+        </div>
+        <div v-if="userIsAuthorized.add_role" class="mave-collection-contributors-subsection">
+          <div class="mave-collection-contributors-subsection-title">Viewers</div>
+          <ul v-if="item.viewers.length > 0">
+            <div v-for="viewer in item.viewers" :key="viewer">
+              <a :href="`https://orcid.org/${viewer.orcidId}`" target="_blank">
+                <img src="@/assets/ORCIDiD_icon.png" alt="ORCIDiD" />
+                {{ viewer.firstName }} {{ viewer.lastName }}
+              </a>
+            </div>
+          </ul>
+          <p v-else>No viewers</p>
+        </div>
+      </div>
+    </div>
+    <div v-else-if="itemStatus=='Loading' || itemStatus=='NotLoaded'">
+      <PageLoading/>
+    </div>
+    <div v-else>
+      <ItemNotFound model="collection" :item-id="itemId"/>
+    </div>
+  </DefaultLayout>
+</template>
+
+<script>
+import axios from 'axios'
+import Button from 'primevue/button'
+import Inplace from 'primevue/inplace'
+import InputText from 'primevue/inputtext'
+import Textarea from 'primevue/textarea'
+
+import EntityLink from '@/components/common/EntityLink'
+import CollectionBadge from '@/components/CollectionBadge'
+import CollectionDataSetEditor from '@/components/CollectionDataSetEditor'
+import CollectionPermissionsEditor from '@/components/CollectionPermissionsEditor'
+import DefaultLayout from '@/components/layout/DefaultLayout'
+import ItemNotFound from '@/components/common/ItemNotFound'
+import PageLoading from '@/components/common/PageLoading'
+import config from '@/config'
+import useAuth from '@/composition/auth'
+import useFormatters from '@/composition/formatters'
+import useItem from '@/composition/item'
+
+export default {
+  name: 'CollectionView',
+
+  components: { Button, CollectionBadge, CollectionDataSetEditor, CollectionPermissionsEditor, DefaultLayout, EntityLink, Inplace, InputText, ItemNotFound, PageLoading, Textarea },
+
+  props: {
+    itemId: {
+      type: String,
+      required: true
+    }
+  },
+
+  data: () => ({
+    userIsAuthorized: {
+      delete: false,
+      publish: false,
+      update: false,
+      add_score_set: false, // permissions are the same for add score set, remove score set, add experiment, and remove experiment
+      add_role: false, // permissions are the same for add user to role and remove user from role
+    },
+    editName: null,
+    displayCollectionNameEdit: false,
+    editDescription: null,
+    displayCollectionDescriptionEdit: false,
+  }),
+
+  setup: () => {
+    const {userIsAuthenticated} = useAuth()
+    return {
+      config: config,
+      userIsAuthenticated,
+
+      ...useFormatters(),
+      ...useItem({ itemTypeName: 'collection' }),
+    }
+  },
+
+  mounted: async function() {
+    await this.checkUserAuthorization()
```

Comment:
Is there a reason we need this intermediary function and can't just await `checkAuthorization` directly?